execution_engine/src/storage/trie_store/operations/tests/scan.rs

```diff
@@ -50,16 +57,22 @@ where
             Trie::Node { pointer_block } => {
                 let pointer_tip_hash = pointer_block[<usize>::from(index)].map(|ptr| *ptr.hash());
                 assert_eq!(Some(expected_tip_hash), pointer_tip_hash);
-                tip = either::Either::Right(Trie::Node { pointer_block });
+                tip = LazilyDeserializedTrie::Node { pointer_block };
             }
             Trie::Extension { affix, pointer } => {
                 let pointer_tip_hash = pointer.hash().to_owned();
                 assert_eq!(expected_tip_hash, pointer_tip_hash);
-                tip = either::Either::Right(Trie::Extension { affix, pointer });
+                tip = LazilyDeserializedTrie::Extension { affix, pointer };
             }
         }
     }
-    assert_eq!(root, tip.expect_right("Unexpected leaf found"));
+
+    assert_matches!(
+        tip,
+        LazilyDeserializedTrie::Node { .. } | LazilyDeserializedTrie::Extension { .. },
+        "Unexpected leaf found"
+    );
+    assert_eq!(root, tip.try_into()?);
```

Comment:
```suggestion
  assert!(matches!(
    tip,
    LazilyDeserializedTrie::Node { .. } | LazilyDeserializedTrie::Extension { .. }),
    "Unexpected leaf found"));
    assert_eq!(root, tip.try_into()?);
```

Reply:
Fixed in https://github.com/casper-network/casper-node/pull/4083/commits/765677f7c07be92bd21cbfd3303dfcf9e74cd819